### ISSUE TITLE
getRealSize should expect also literals, not only integers

### ISSUE THREAD

**Issue by iamntz** [phpmyadmin/phpmyadmin#14307]
```php
 if ($postsize = ini_get('post_max_size')) {
            $this->set(
                'max_upload_size',
                min(Core::getRealSize($filesize), Core::getRealSize($postsize))
            );
        } else {
            $this->set('max_upload_size', Core::getRealSize($filesize));
        }
```
This block throw an error, because getRealSize is failing if the php.ini will have shorthands (e.g. 1G).

The default `php.ini` is  `post_max_size=8M`

### Server configuration
**Operating system**:
Windows 10

**Web server:**

**Database:**
```
mysql --version
mysql  Ver 15.1 Distrib 10.2.14-MariaDB, for Win64 (AMD64)
```

**PHP version:**
```
PHP 7.2.5 (cli) (built: Apr 25 2018 02:39:21) ( ZTS MSVC15 (Visual C++ 2017) x86 )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Xdebug v2.6.0, Copyright (c) 2002-2018, by Derick Rethans
```

**phpMyAdmin version:**
HEAD

**Comment by nulll-pointer** [comment]
> This block throw an error, because getRealSize expects strictly an integer, but will fail if the php.ini will have shorthands (e.g. 1G).

I don't think it's true, Having a quick look at the function definition, I think it accepts the shorthands.
If the size is not in the shorthand format, it typecasts it into an integer.

**Comment by iamntz** [comment]
Yeah, so it seems I'm a bit dumb about this new return type declaration thingy, so I misunderstood some stuff. That `: int` at the end of the declaration means that the method should _return_ an integer, and not that is _expecting_ an integer (dooh).

---

I'm not sure if this is specific on Windows platform, on my machine or PHP in general, but stuff looks like this. I did a bit of fidding on `Core` class, on method `getRealSize`

```
gettype($matches[1]) => string (value is 2)
gettype($binaryprefixes[$matches[2]]) => integer
gettype($matches[1] * $binaryprefixes[$matches[2]])  => double (value is 2147483648)
```
Now, this gets _really_ interesting:

        if (preg_match('/^([0-9]+)([KMGT])/i', $size, $matches)) {
            return (int) ($matches[1] * $binaryprefixes[$matches[2]]);
        }


If I'm doing this, the return value is `-2147483648`. Yup, negative.

Can anyone explain why this is happen? I mean the confusion is double:

- why multiplying an integer with a string results in an double?
- why casting to `int` converts to negative?

**Comment by nulll-pointer** [comment]
It's good that you mentioned that you are using windows, as stated [here](https://stackoverflow.com/questions/670662/whats-the-maximum-size-for-an-int-in-php), the size of int in windows below php 7 is 32 bits, maybe that's why it [overflows](https://en.wikipedia.org/wiki/Integer_overflow).

**Comment by williamdes** [comment]
@iamntz What do you expect to be done to fix this issue ?
Is this a phpMyAdmin issue ?

**Comment by iamntz** [comment]
@williamdes I guess is more of an architecture issue (i.e. x86 vs x64) rather than phpMyAdmin.

But since PMA isn't tied to a certain architecture, i guess it might deal with this edge case.

**Comment by williamdes** [comment]
Probably fixed by efb5bfdc89ec3be45d748347c0886cc37efdf637
Ref: #15926

I am not sure we can do anything for 32 bit platforms
